src/Dialog/ConfirmationDialog.tsx

```diff
@@ -157,7 +172,7 @@ export const ConfirmationDialog: React.FC<ConfirmationDialogProps> = props => {
     variant: 'large',
     onClick: onConfirmButtonClick
   }
-  const footerButtons = [cancelButton, confirmButton]
+  const footerButtons = [cancelButton, confirmButton].filter(btn => !!btn.content)
```

Comment:
I could also add a `cancelButtonContentDisabled` prop with a default boolean value set to false to make the prop optional, rather than checking for existence. I didn't want to make a big change before consulting with the team, but definitely open to suggestions 🙂